api/turing/cluster/servicebuilder/service_builder.go

```diff
@@ -353,18 +353,23 @@ func (sb *clusterSvcBuilder) NewSecret(
 }
 
 // NewPodDisruptionBudget creates a new `cluster.PodDisruptionBudget`
-// for the given service (router/enricher/ensembler).
+// for the given service (router/enricher/ensembler/fluentd logger).
 func (sb *clusterSvcBuilder) NewPodDisruptionBudget(
 	routerVersion *models.RouterVersion,
 	project *mlp.Project,
 	componentType string,
 	pdbConfig config.PodDisruptionBudgetConfig,
 ) *cluster.PodDisruptionBudget {
+	var appNameSuffix string
+	if componentType != ComponentTypes.FluentdLogger {
+		appNameSuffix = "-0"
+	}
```

Comment:
This is needed because the Fluentd stateful set doesn't have a revision number as the suffix in its app name, which is the case for Knative apps (Turing router/enricher/ensembler).

Reply:
this got me thinking if the others should **not** be suffix with -0. During any patching, the suffix will not stay as `-0`and the pdb selector will not pick up anything

should we be using other key instead of `app` that will be consistent for the deployment even after patching?
something like 
`serving.knative.dev/service=abc-turing-ensembler-9` 
`serving.knative.dev/service=abc-turing-router-9` 
 `app=abc-turing-fluentd-logger-9`
 
 @ariefrahmansyah  any thoughts?

Reply:
Hmm, seems like we can take away the revision number for the label that has `app` as the key i.e. use `app=abc-turing-fluentd-logger` instead (the other labels can remain the same).

Reply:
I just checked that `app` and `serving.knative.dev/revision` labels also have `-0` suffix for Turing router. But you can use `serving.knative.dev/service` as this one does not have the suffix. It should be fine using other label(s) that does not have `-0` suffix.

Reply:
Thanks for the suggestion! :D Added that to the changes in this PR!